pkg/nixbase32/nixbase32.go

```diff
@@ -8,6 +8,42 @@ import (
 // Alphabet contains the list of valid characters for the Nix base32 alphabet.
 const Alphabet = "0123456789abcdfghijklmnpqrsvwxyz"
 
+// ValidateString validates if a byte slice is valid nixbase32.
+func Validate(b []byte) error {
+	return ValidateString(string(b))
+}
+
+// ValidateString validates if a string is valid nixbase32.
+func ValidateString(s string) error {
+	dstLen := DecodedLen(len(s))
+
+	for n := 0; n < len(s); n++ {
+		c := s[len(s)-n-1]
+
+		digit := strings.IndexByte(Alphabet, c)
+		if digit == -1 {
+			return fmt.Errorf("character %v not in alphabet", c)
+		}
+
+		b := uint(n * 5)
+		i := b / 8
+		j := b % 8
+
+		// calculate the "carry pattern"
+		carry := byte(digit) >> (8 - j)
+
+		// if we're at the end of dst…
+		if i == uint(dstLen-1) {
+			// but have a nonzero carry, the encoding is invalid.
+			if carry != 0 {
+				return fmt.Errorf("invalid encoding")
+			}
+		}
+	}
```

Comment:
This is pretty much copy-pasted from `DecodeString`. Can we move this into a private function and use it from both places?

Reply:
Done.